shacl12-ui/index.html

```diff
@@ -801,8 +801,592 @@ <h3>Document Conventions</h3>
+            <b>Score:</b>
+            10 for <code>xsd:dateTime</code> literals.
+            5 if the property has <code>sh:datatype xsd:dateTime</code>.
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a calendar-like date picker including a time selector.
+          </p>
+          <img src="images/editors/DateTimePickerEditor.png" />
+          <pre class="example-shapes">
+ex:Customer-lastVisitTime
+	a sh:PropertyShape ;
+	sh:path ex:lastVisitTime ;
+	<b>sh:datatype xsd:dateTime ;</b>
+	...</pre>
+        </section>
+        <section id="DetailsEditor">
+          <h4>shui:DetailsEditor</h4>
+          <p>
+            <b>Score:</b>
+            <code>null</code> for non-literals, i.e. it can be selected manually via <code>shui:editor</code>.
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            typically rendering as a nested form, using the properties defined by the <code>sh:node</code> or <code>sh:class</code> (in that order)
+            of the property as fields.
+            Alternative renderings are possible, such as opening the resource in a separate dialog.
+          </p>
+          <p>
+            This is particularly useful for some types of blank nodes that only make sense to be edited in the context of their parent resource.
+          </p>
+          <img src="images/editors/DetailsEditor.png" width="782" height="364" />
+          <pre class="example-shapes">
+ex:Product
+	a owl:Class ;
+	a sh:NodeShape ;
+	rdfs:label "Product" ;
+	rdfs:subClassOf owl:Thing ;
+	sh:property ex:Product-weight .
+
+ex:Product-weight
+	a sh:PropertyShape ;
+	sh:path ex:weight ;
+	shui:editor shui:DetailsEditor ;
+	shui:viewer shui:DetailsViewer ;
+	sh:description "A blank node with a numeric field and a unit which is one of the QUDT mass units." ;
+	sh:maxCount 1 ;
+	sh:name "weight" ;
+	sh:node ex:ValueWithWeight ;
+	sh:nodeKind sh:BlankNode .
+
+ex:ValueWithWeight
+	a sh:NodeShape ;
+	rdfs:label "Value with weight" ;
+	sh:property ex:ValueWithWeight-numericValue ;
+	sh:property ex:ValueWithWeight-unit .
+
+ex:ValueWithWeight-numericValue
+	a sh:PropertyShape ;
+	sh:path ex:numericValue ;
+	sh:datatype xsd:decimal ;
+	sh:maxCount 1 ;
+	sh:minCount 1 ;
+	sh:name "numeric value" .
+
+ex:ValueWithWeight-unit
+	a sh:PropertyShape ;
+	sh:path ex:unit ;
+	sh:class &lt;http://qudt.org/schema/qudt/Unit> ;
+	sh:maxCount 1 ;
+	sh:minCount 1 ;
+	sh:name "unit" ;
+	sh:node [
+		rdfs:label "Permissible values must have quantity kind Mass." ;
+		sh:property [
+			sh:path &lt;http://qudt.org/schema/qudt/hasQuantityKind> ;
+			sh:hasValue &lt;http://qudt.org/vocab/quantitykind/Mass> ;
+		] ;
+	] .</pre>
+          <p>
+            This widget requires that the surrounding property (ex:weight above) declares sh:nodeKind sh:BlankNode
+            and also has a sh:node constraint that points at a node shape that declares the properties that shall be editable.
```

Comment:
```suggestion
            This widget requires that the surrounding property (<code>ex:weight</code> above) declares <code>sh:nodeKind sh:BlankNode</code>
            and also has a <code>sh:node</code> constraint that points at a node shape that declares the properties that shall be editable.
```

Reply:
Just a comma, missing from @smessie's suggestion
```suggestion
            This widget requires that the surrounding property (<code>ex:weight</code>, above) declares <code>sh:nodeKind sh:BlankNode</code>
            and also has a <code>sh:node</code> constraint that points at a node shape that declares the properties that shall be editable.
```